backend/src/test/java/gov/cdc/usds/simplereport/api/pxp/PatientExperienceControllerTest.java

```diff
@@ -138,6 +138,87 @@ void verifyLinkReturnsPerson() throws Exception {
         .andExpect(jsonPath("$.lastName", is(_person.getLastName())));
   }
 
+  @Test
+  void verifyLinkReturns410forExpiredLinks() throws Exception {
+    // GIVEN
+    // TODO: I need a slightly different test setup here, and idk how to do this
+    // with the way that the @BeforeEach is being transactionalized
+    _truncator.truncateAll();
```

Comment:
Sooo lines 146-153 here are the same as the `@BeforeEach` block except that on 153 I call `createExpiredPatientLink` instead of `createPatientLink`. This implementation means that this particular spec truncates the DB _twice_, which is... dumb. How can I do this better without creating a differently configured class for the expiry case?

The problem is really the transactional nature of the DB interactions – I can't just call `_patientLink.expire()` and save it to the repository because there's no Session

cc: @benwarfield-usds ?

Reply:
I see two less-annoying options:

1. add a method to TestDataFactory called `expireLink` that does what you want but wrapped in a `@Transactional` annotation
2. create an expired link and an unexpired link in the beforeEach initializer.

Looking at it more closely, the second one seems pretty annoying, since you'd like to only have one patient link active per patient test order (though _technically_ I don't believe that is enforced), so the first seems like the simplest workaround.

Reply:
The first option worked great!